app/common/lib/pinnedSitesUtil.js

```diff
@@ -110,13 +110,16 @@ const getKey = (siteDetail) => {
   if (!siteDetail) {
     return null
   }
+  const location = siteDetail.get('location')
+  const partitionNumber = siteDetail.get('partitionNumber', 0)
+  return getKeyFromLocation(location, partitionNumber)
+}
 
-  let location = siteDetail.get('location')
+const getKeyFromLocation = (location, partitionNumber) => {
```

Comment:
Not sure why we need to move this in to the new function. You could just create Immutable.Map() in your test (as far as I can see is that the only place that you use it).

```
Immutable.fromJS({
  location: 'something',
  partitionNumber: 1
})
```

Reply:
@NejcZdovc the aim here was to avoid having 2+ separate places where this format of string 'key' for pinned tabs is defined (`${location}|${parititionNumber}`). The action `onPinnedTabReorder` requires this key, the generation of which which was previously hidden inside another function.
Here, I expose the `getKey` function from one place, so that if we want to change that format (perhaps introduce a new property, or change the separator charactor, etc) it is controlled by a single function.
Now that I've explained the why, I'm happy to change it - maybe I can get it from the existing state, or the DOM, or something...  ☺️

Reply:
Do we have it defined in two places? :smiley: I am not aware of it. For pinnedSites we have only one getKey and format AFAIK

Reply:
Well there would be if I added a new place here 😀 , but I've taken your advice and instead made copies of the 'SiteDetail' structure the original function requires.